pkg/collector/corechecks/gpu/nvmlmetrics/collector.go

```diff
@@ -12,19 +12,24 @@
 package nvmlmetrics
 
 import (
+	"errors"
 	"fmt"
 
 	"github.com/NVIDIA/go-nvml/pkg/nvml"
 	"github.com/hashicorp/go-multierror"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
 // Collector is the main struct responsible for collecting metrics from the NVML library.
 type Collector struct {
 	lib        nvml.Interface
-	collectors []subsystemCollector
-	devices    []nvml.Device
+	collectors map[nvml.Device][]subsystemCollector
```

Comment:
According to [this](https://go.dev/ref/spec#Comparison_operators), "Interface types that are not type parameters are comparable. Two interface values are equal if they have [identical](https://go.dev/ref/spec#Type_identity) dynamic types and equal dynamic values or if both have value nil", it seems the fields of the interface don't matter.

Reply:
Why does Go have to be so weird :( I thought they were comparing the underlying pointers and checking if they were the same. Do you think it's better to change the type and use another struct or check if it works with nvml.Device?

Reply:
i saw a few [places](https://forum.golangbridge.org/t/using-interface-types-as-map-keys/706) where they recommend to avoid using interface as the key for golang maps but i wouldn't call them very "official" sources. 

i raised this because in our scenario the runtime panic is dependent on 3rd party code (nvml.Device implementation in go-nvml library)